elk/extraction/extraction.py

```diff
@@ -148,24 +135,69 @@ def extract_hiddens(
             )
             for layer_idx in layer_indices
         }
+        lm_preds = torch.empty(
+            num_variants,
+            2,  # contrast pair
+            device=device,
+            dtype=torch.float32,
+        )
         text_inputs = []
 
         # Iterate over variants
         for i, record in enumerate(example["prompts"]):
             variant_inputs = []
 
             # Iterate over answers
-            for j in range(2):
-                text = record[j]["text"]
+            for j, choice in enumerate(record):
+                text = choice["text"]
                 variant_inputs.append(text)
 
+                # TODO: Do something smarter than "rindex" here. Really we want to
+                # get the span of the answer directly from Jinja, but that doesn't
+                # seem possible. This approach may fail for complex templates.
+                answer_start = text.rindex(choice["answer"])
+
+                # Only feed question, not the answer, to the encoder for enc-dec models
+                if model.config.is_encoder_decoder:
+                    # TODO: Maybe make this more generic for complex templates?
+                    text = text[:answer_start].rstrip()
```

Comment:
Why do we strip the right here? This kind of assumes the answer string starts with whitespace correct?

Reply:
No, we don't assume that. It's not the case that the concatenation of the encoder input and the decoder input have to like, make sense or be a full sentence or something. I'm sort of assuming that the answer string does _not_ start with whitespace. And the encoder is not going to expect trailing whitespace in its input (although I doubt it actually matters)